checkout.go

```diff
@@ -65,6 +66,15 @@ func populateCheckoutOpts(ptr *C.git_checkout_options, opts *CheckoutOpts) *C.gi
 	if opts.TargetDirectory != "" {
 		ptr.target_directory = C.CString(opts.TargetDirectory)
 	}
+	if len(opts.Paths) > 0 {
+		var pathArray []*C.char = make([]*C.char, len(opts.Paths))
+		for i, s := range opts.Paths {
+			pathArray[i] = C.CString(s)
+		}
+		ptr.paths.strings = (**C.char) (unsafe.Pointer(&pathArray[0]))
+		ptr.paths.count = C.size_t(len(opts.Paths))
+	}
+
```

Comment:
We have a function so we don't have to write this code every time, use `makeCstringsFromString()`.